django_redis/hash_ring.py

```diff
@@ -25,33 +28,35 @@ def add_node(self, node):
 
         self.sorted_keys.sort()
 
-    def remove_node(self, node):
+    def remove_node(self, node: str):
         self.nodes.remove(node)
         for x in range(self.replicas):
-            _hash = hashlib.sha256("%s:%d" % (node, x)).hexdigest()
-            self.ring.remove(_hash)
+            _hash = hashlib.sha256(f"{node}:{x}".encode()).hexdigest()
+            del self.ring[_hash]
             self.sorted_keys.remove(_hash)
 
-    def get_node(self, key):
+    def get_node(self, key) -> Optional[str]:
         n, i = self.get_node_pos(key)
         return n
 
-    def get_node_pos(self, key):
+    def get_node_pos(self, key: str) -> Tuple[Optional[str], Optional[int]]:
         if len(self.ring) == 0:
-            return (None, None)
+            return None, None
 
         _hash = hashlib.sha256(key.encode()).hexdigest()
         idx = bisect.bisect(self.sorted_keys, _hash)
         idx = min(idx - 1, (self.replicas * len(self.nodes)) - 1)
-        return (self.ring[self.sorted_keys[idx]], idx)
+        return self.ring[self.sorted_keys[idx]], idx
 
-    def iter_nodes(self, key):
+    def iter_nodes(
+        self, key: str
+    ) -> Generator[Tuple[Optional[str], Optional[str]], str, None]:
```

Comment:
This is probably the more compact form (see the last example here: https://docs.python.org/3/library/typing.html#typing.Generator)

```suggestion
    ) -> Iterator[Tuple[Optional[str], Optional[str]]]:
```

Reply:
But it has a `SendType` so Iterator is not really describing the `Generator`.
It is arguable that default to `None` is not a good returning type, typingwise

Reply:
But the send wouldn't be used here, right? I'm aware of the difference I just don't think it matters for this code